### PR TITLE
Support missing alg in Jwks

### PR DESCRIPTION
**Pull request by qiwzhang**
According to https://tools.ietf.org/html/rfc7517#section-4.4

"alg" is JWKS is optional.

### CODE REVIEW

File: src/jwks.cc [hunk 1]
```diff
@@ -184,16 +182,24 @@ Status extractJwkFromJwkEC(const ::google::protobuf::Struct& jwk_pb,
...
-             (jwk->crv_ == "P-384" || jwk->crv_ == "")) {
+  } else if (jwk->alg_ == "ES384" || jwk->crv_ == "P-384") {
     nid = NID_secp384r1;
     jwk->crv_ = "P-384";
-  } else if (jwk->alg_ == "ES512" &&
-             (jwk->crv_ == "P-521" || jwk->crv_ == "")) {
+  } else if (jwk->alg_ == "ES512" || jwk->crv_ == "P-521") {
     nid = NID_secp521r1;
     jwk->crv_ = "P-521";
   } else {
```
Comment (yangminzhu) [inline src/jwks.cc]: Can we now return a different error in this case? Maybe JwksECKeyAlgOrCrvUnsupported?

File: src/jwks.cc [hunk 2]
```diff
@@ -184,16 +182,24 @@ Status extractJwkFromJwkEC(const ::google::protobuf::Struct& jwk_pb,
...
+    if (!((jwk->alg_ == "ES256" && jwk->crv_ == "P-256") ||
+          (jwk->alg_ == "ES384" && jwk->crv_ == "P-384") ||
+          (jwk->alg_ == "ES512" && jwk->crv_ == "P-521"))) {
+      return Status::JwksECKeyAlgNotCompatibleWithCrv;
+    }
+  }
+
   int nid;
-  if (jwk->alg_ == "ES256" && (jwk->crv_ == "P-256" || jwk->crv_ == "")) {
+  if (jwk->alg_ == "ES256" || jwk->crv_ == "P-256") {
```
Comment (yangminzhu) [inline src/jwks.cc]: is the `crv` required in the JWK? The behavior changes if both `alg` and `crv` is not set after this PR, is this expected?

File: src/jwks.cc (hunk 1 above)
Comment (qiwzhang) [reply src/jwks.cc]: Done

File: src/jwks.cc (hunk 2 above)
Comment (qiwzhang) [reply src/jwks.cc]: Added the default handling, not to break existing customers.

**Pull request merged**